### DISCUSSION TITLE
Issues with requirements in Anaconda environment?

### DISCUSSION THREAD

**Discussion by Joerg0** [sebastian-hofstaetter/teaching#130] [Advanced IR: Exercise Q&A] [answered]
Maybe I'm the only one who has this problem, maybe someone has a tip for me: I create an anaconda environment with Py3.6 in PyCharm (because torch needs exactly py3.6) and try to install the environments.
Neither from PyCharm nor through the terminal I succeed, I get e.g.:

PackagesNotFoundError: The following packages are not available from current channels:
   - blingfire==0.1.7
 
Any idea how to install the requirements? Thanks in advance!

**Comment by pintersebastian** [comment]
Hi!
I had similiar issues, but managed to fix it, but i dont remember how exactly. I will try to recreate.

Did you install via anaconda navigator or via console? 
Did you try via pip?

Kind Regards,
Sebastian

**Comment by Joerg0** [comment, marked as answer]
I am on Win11.

0.) I added C:\ProgramData\Anaconda3\Scripts to path in the system variables 

1.) I started Powershell with Admin rights and updated conda with
conda update -n base -c defaults conda

2.) With
conda create -n AIRenvCONDApy36 python=3.6
I created the environment in C:\Users\username\.conda\envs\AIRenvCONDApy36

3.) conda activate AIRenvCONDApy36

4.) cd 'D:\LinkToPyProjects\air-2023-group24'

5A) pip install -r requirements.txt
returns:
	ERROR: Could not find a version that satisfies the requirement torch==1.6.0 (from versions: 1.7.0, 1.7.1, 1.8.0, 1.8.1, 1.9.0, 1.9.1, 1.10.0, 1.10.1, 1.10.2)
ERROR: No matching distribution found for torch==1.6.0

5B) If I delete the row 
torch==1.6.0
in requirements.txt and run
	pip install -r requirements.txt
the rest installs without problems.

From
	pip list
I then get the info that I have
	torch               1.7.1

6.) With 
	conda install pytorch==1.6.0 torchvision==0.7.0 cpuonly -c pytorch
it eventually installs the last requirement.


Okay, so far so fine. I get back to PyCharm, open the project, add the conda environment, check the requirements and it turns out that everything is fine except torch, which is still 1.7.1

I tried to manually install the 1.6.0 package in the Python interpreter, but it remains 1.7.1.

Eventually with
pip install torch==1.6.0+cpu torchvision==0.7.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
it worked.

I only have a virtual GPU - so I took 1.6.0+cpu, hope this will be okay - ah okay, doesn't matter probably because the training will be done on colab's GPU.

**Reply by mandoway** [reply]
This worked for me!
Thanks!

Tried it on also on Mac but had no luck with that, maybe someone has an idea for that?

Here is also a TL;DR version (windows 11):  

0. Install python 3.6 & create venv (I used pyCharm to create a venv)
1. Remove torch from requirements
2. Install requirements 
3. Run 
`pip install torch==1.6.0+cu101 torchvision==0.7.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html`
 to install the exact torch version (swap the +cu101 for appropriate cuda version or +cpu)
